выполнена работа по 5 спринту

Fensiz opened the pull request.

Fensiz committed:
логика формирования вопросов вынесена в отдельный класс

Fensiz committed:
выполнена работа по 5 спринту

Fensiz committed:
мелкие правки: убраны лишние табуляции и изменен порядок свойств/методов

ilnurMugaev commented on MovieQuiz/Helpers/UserDefaults+Extensions.swift:
```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+extension UserDefaults {
+	enum Key: String {
+		case gamesCount, bestCorrect, bestTotal, bestDate, totalCorrect, total
+	}
+
+	// Сохранение значения с использованием enum
+	func set(_ value: Int, forKey key: Key) {
+		self.set(value, forKey: key.rawValue)
+	}
+	func set(_ value: Any?, forKey key: Key) {
+		self.set(value, forKey: key.rawValue)
+	}
+
+	// Получение значения с использованием enum
+	func integer(forKey key: Key) -> Int {
+		return self.integer(forKey: key.rawValue)
+	}
+	func object(forKey key: Key) -> Any? {
+		return self.object(forKey: key.rawValue)
+	}
```
После автоформатирования (`Control + I`) изменились отступы. Рекомендуется использовать 4 пробела для отступов и выбрать использование пробелов вместо табуляции в настройках Xcode. Это предотвратит мерж-конфликты и улучшит читаемость кода в совместной разработке. Проверь весь код и всегда делай `Cmd + A` и `Control + I` перед сохранением

Fensiz replied:
Сделал Cmd + A и Control + I - изменений в этом файле не произошло, остальные проверю, спасибо

ilnurMugaev commented on MovieQuiz/Models/QuizQuestion.swift:
```diff
@@ -0,0 +1,16 @@
+//
+
+struct QuizQuestion {
+	// строка с названием фильма,
+	// совпадает с названием картинки афиши фильма в Assets
+	let image: String
+	// строка с вопросом о рейтинге фильма
+	let text: String
+	// булевое значение (true, false), правильный ответ на вопрос
+	let correctAnswer: Bool
```
Рекомендуется избегать излишне подробных комментариев, так как они усложняют чтение кода. Комментарии лучше оставлять в неочевидных местах. Проверь весь код, чтобы убедиться, что комментарии действительно необходимы и полезны

ilnurMugaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -30,60 +31,42 @@ final class MovieQuizViewController: UIViewController {
 	// MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
+		questionFactory = {
+			let questionFactory = QuestionFactory()
+			questionFactory.delegate = self
+			return questionFactory
+		}()
+		alertPresenter = AlertPresenter(controller: self)
```
Повторно присваиваются значения для `alertPresenter` и `questionFactory`, хотя они уже инициализируются при объявлении. Необходимо выбрать одно из решений: либо убрать повторное присваивание, либо оставить переменные без начальной инициализации и инициализировать их в `viewDidLoad`

ilnurMugaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -96,75 +79,48 @@ final class MovieQuizViewController: UIViewController {
-		imageView.image = step.image
-		textLabel.text = step.question
-		counterLabel.text = step.questionNumber
+	private func show(next step: QuizStepViewModel) {
```
Рекомендуется оставлять пустые строки между методами для улучшения читаемости и структурирования кода. Проверь весь код и добавь пустые строки там, где это необходимо

ilnurMugaev commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -0,0 +1,62 @@
+
+import Foundation
+
+class QuestionFactory : QuestionFactoryProtocol {
```
Рекомендую сделать класс `final` для предотвращения наследования и повышения производительности

ilnurMugaev commented on MovieQuiz/Presentation/AlertPresenter.swift:
```diff
@@ -0,0 +1,29 @@
+
+import UIKit
+
+class AlertPresenter : AlertPresenterProtocol {
```
Давай тут тоже добавим модификатор `final` для класса

Fensiz committed:
внесены правки по замечаниям

во всех файла проставлены идентичные отступы
с использованием Cmd + A, Control + I

добавлены интервалы между сложными свойствами и методами

убрано повторное задание фабрики и презентера

сокращены или убраны комментарии

где возможно, классам проставлен модификатор final

убраны лишние import'ы
